### PR TITLE
web3 external provider - Quicknode

### PR DESCRIPTION
**Pull request by jdevcs**
## Description

#6976

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: packages/web3-providers/src/web3_provider_quicknode.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        _host: string) {
+        
+        let host = "";
+        let token = "";
+
+        switch (network) {
+            case Network.ETH_MAINNET:
+                host = isValid(_host) ? _host : "powerful-holy-bush.quiknode.pro";
+                token = isValid(_token) ? _token : "3240624a343867035925ff7561eb60dfdba2a668";
+                break;
```
Comment (avkos) [inline packages/web3-providers/src/web3_provider_quicknode.ts]: is it ok to have tokens here?

**Review by avkos — CHANGES_REQUESTED** [review]

File: packages/web3-providers/package.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
+{
+	"name": "web3-providers",
+	"version": "0.1.0",
```
Comment (luu-alex) [inline packages/web3-providers/package.json]: if we start at 0.1.0 that would that be considered alpha

**Review by luu-alex — COMMENTED** [review]
Looks good, just some feedback.
Integration tests should be created before release of the package

File: packages/web3-providers/src/web3_provider_quicknode.ts (hunk 1 above)
Comment (jdevcs) [reply packages/web3-providers/src/web3_provider_quicknode.ts]: yes, these are public tokens

File: packages/web3-providers/package.json (hunk 2 above)
Comment (jdevcs) [reply packages/web3-providers/package.json]: yes, 1.0.0 will be prod

**Review by avkos — APPROVED** [review]

**Comment by luu-alex** [comment]
Failing testcases, otherwise it looks good

**Pull request merged**